vm/systemSmartContracts/delegation.go

```diff
@@ -1215,6 +1215,12 @@ func (d *delegation) stakeNodes(args *vmcommon.ContractCallInput) vmcommon.Retur
 		return vmOutput.ReturnCode
 	}
 
+	allLogs := d.eei.GetLogs()
+	if tooManyNodesLogs(allLogs) {
```

Comment:
is there no way to see that the number of nodes it too high from the vmOutput or the error? Looking in the logs seems more like a "hack".

Also I think an unit tests is missing.